Applications/SystemMonitor/main.cpp

```diff
@@ -342,6 +349,23 @@ int main(int argc, char** argv)
 
     window->set_icon(app_icon.bitmap_for_size(16));
 
+    if (args_tab_view == "processes")
+        tabwidget.set_tab_index(0);
+    else if (args_tab_view == "graphs")
+        tabwidget.set_tab_index(1);
+    else if (args_tab_view == "fs")
+        tabwidget.set_tab_index(2);
+    else if (args_tab_view == "pci")
+        tabwidget.set_tab_index(3);
+    else if (args_tab_view == "devices")
+        tabwidget.set_tab_index(4);
+    else if (args_tab_view == "network")
+        tabwidget.set_tab_index(5);
+    else if (args_tab_view == "processors")
+        tabwidget.set_tab_index(6);
+    else if (args_tab_view == "interrupts")
+        tabwidget.set_tab_index(7);
```

Comment:
Let's not hard-code the index of the tabs like this, or it's easy to make this buggy when adding/removing/moving tabs in the future.
Instead, let's hold on to pointers to the various tab page widgets when we create them and use `set_active_widget()` instead. :)

Reply:
I got all to work with pointers now but the processes tab which is added using `tabwidget.add_tab` and not `add_widget` has it's pointer done a bit weirdly using `auto processes_tab = tabwidget.active_widget()` because i'm not sure rightnow how can i get a GUI::Widget instead of getting GUI::VerticalSplitter.

And i want to ask, should i squash it _(somehow)_ to the original SystemMonitor commit? _(i still suck at git so i'll rather try messing with squashing commits somewhere else so i don't mess it up later )_

Reply:
**GUI::VerticalSplitter** inherits from **GUI::Widget**, so the pointers are already compatible :)